### ISSUE TITLE
Expose shows clickable links before commit

### ISSUE THREAD

**Issue by galgalesh** [juju/juju-gui#2666]
Unexposed Charm: 

![image](https://cloud.githubusercontent.com/assets/1492981/24199893/87bbbee6-0f0b-11e7-955e-280dbb666fb5.png)

When you click on expose, a new field appears with the Charm's public IP as a clickable link.

![image](https://cloud.githubusercontent.com/assets/1492981/24199929/aacc3636-0f0b-11e7-906b-5e99cbf6bff6.png)

This makes me think that the application is exposed. I click on the link and my browser times out. This is because the application isn't exposed until I click "Commit Changes".

**Comment by frankban** [comment]
Good catch! We should not show that as a link before the charm is actually exposed. Thanks.

**Comment by frankban** [comment]
Note: this requires us to register and keep track of whether the app is exposed in Juju even after a change has been made but not committed by a GUI user.

**Comment by cassiocassio** [comment]
Hmmm. Normal human beings might imagine that clicking the button performs the action. 

Also:

`$ juju expose wordpress`

would do that immediately, right?

There're a set of large-scale actions, where the [ commit changes ] , review and confirm, makes some workflow-sense.

There's another set of actions where, surely,  I should be able to click and be done with it.

We only have one interaction model for performing changes in the GUI. So everything goes through the big button.

**Comment by hatched** [comment]
@cassiocassio yes all CLI commands happen immediately. 

There used to be a bug in the GUI in which the expose was triggered immediately but this was fixed in interested of having a consistent flow.
